rustfst-ffi/src/iterators.rs

```diff
@@ -0,0 +1,205 @@
+use crate::fst::CFst;
+use crate::tr::CTr;
+use crate::{get, get_mut, wrap, CStateId, RUSTFST_FFI_RESULT};
+use anyhow::{anyhow, Result};
+use ffi_convert::*;
+use rustfst::fst_traits::CoreFst;
+use rustfst::prelude::{StateId, StateIterator, Tr, TropicalWeight, VectorFst};
+
+use std::iter::Peekable;
+use std::ops::Range;
+
+#[derive(Debug)]
+pub struct TrsIterator<'a, F: CoreFst<TropicalWeight>> {
+    fst: &'a F,
```

Comment:
We might be able to store the `Trs` object directly in this iterator to avoid the lookup at each iteration